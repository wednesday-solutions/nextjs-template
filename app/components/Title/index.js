/**
 *
 * Title
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import StarIcon from '@mui/icons-material/Star';
import Text from '@app/components/Text/index';
import fonts from '@app/themes/fonts';
import { Box, Skeleton } from '@mui/material';
import If from '../If/index';

function Title(props) {
  const { name, loading, stargazersCount } = props;
  const headingStyle = fonts.style.heading();

  const renderSkeleton = () => {
    return (
      <>
        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
      </>
    );
  };

  return (
    <Box sx={{ justifyContent: 'space-between', alignItems: 'center' }} data-testid="title">
      <If condition={loading} otherwise={renderSkeleton()}>
        <Text styles={headingStyle}>{name}</Text>
        <Text styles={headingStyle}>
          <StarIcon /> ( {stargazersCount} )
        </Text>
      </If>
    </Box>
  );
}

Title.propTypes = {
  loading: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  stargazersCount: PropTypes.number.isRequired
};

export default Title;
