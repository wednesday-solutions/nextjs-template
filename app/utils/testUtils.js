import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../configureStore';
import colors from '@themes/colors';
import { translationMessages, DEFAULT_LOCALE } from '../i18n';

export const getComponentStyles = (Component, props = {}) => {
  renderProvider(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);

  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = (children) => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={colors}>{children}</ThemeProvider>
    </IntlProvider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const renderStoreProvider = (children) => {
  const store = configureStore({});
  return render(
    <Provider store={store}>
      <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
        <ThemeProvider theme={colors}>{children}</ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};
