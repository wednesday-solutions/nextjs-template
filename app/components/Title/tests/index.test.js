/**
 *
 * Tests for Title
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderProvider } from '@utils/testUtils'
import Title from '../index'

describe('<Title />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Title />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 Title component', () => {
    const { getAllByTestId } = renderProvider(<Title />)
    expect(getAllByTestId('title').length).toBe(1)
  })
})
