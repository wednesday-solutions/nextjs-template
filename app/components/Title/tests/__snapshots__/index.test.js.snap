// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render and match the snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  50% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }

  100% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: block;
  background-color: rgba(0, 0, 0, 0.11);
  height: auto;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px/6.7px;
  position: relative;
  overflow: hidden;
  -webkit-mask-image: -webkit-radial-gradient(white, black);
}

.emotion-1:empty:before {
  content: "\\00a0";
}

.emotion-1::after {
  -webkit-animation: animation-0 1.6s linear 0.5s infinite;
  animation: animation-0 1.6s linear 0.5s infinite;
  background: linear-gradient(
            90deg,
            transparent,
            rgba(0, 0, 0, 0.04),
            transparent
          );
  content: '';
  position: absolute;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
}

<body>
  <div>
    <div
      class="MuiBox-root emotion-0"
      data-testid="title"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave emotion-1"
        data-testid="skeleton"
        style="height: 40px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave emotion-1"
        data-testid="skeleton"
        style="height: 40px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave emotion-1"
        data-testid="skeleton"
        style="height: 40px;"
      />
    </div>
  </div>
</body>
`;
